extensions-core/lookups-cached-single/src/main/java/io/druid/server/lookup/jdbc/JdbcDataFetcher.java

```
@@ -190,10 +262,41 @@ public boolean equals(Object o)
     if (!keyColumn.equals(that.keyColumn)) {
       return false;
     }
+    if (!prefetchKeyProvider.equals(that.prefetchKeyProvider)) {
+      return false;
+    }
     return valueColumn.equals(that.valueColumn);
 
   }
 
+  private String makePrefetchQuery(String key)
+  {
+    PrefetchKeyProvider.ReturnType returnType = prefetchKeyProvider.getReturnType();
+    String[] keys = prefetchKeyProvider.get(key);
+
+    switch(returnType) {
+      case Range:
+        Preconditions.checkArgument(keys.length == 2,
+            "Needs only two points of range (start and end) but got %d", keys.length);
+        String from = keys[0];
+        String to = keys[1];
+        return
+            (from == null) ? String.format(prefetchQueryToOnly, to) :
+                (to == null) ? String.format(prefetchQueryFromOnly, from)
+                             : String.format(prefetchQueryFromTo, from, to);
+      case Points:
+        Preconditions.checkArgument(keys.length > 0, "Needs at least one key to prefetch");
+        String inList = String.format(
+            "'%s'",
+            StringUtils.join(keys, "', '")
+            );
+        return String.format(prefetchQueryForKeys, inList);
+    }
+
+    // should not reach here
```

what does this mean ? should we raise an exception ?